runtime/3rdparty/ggml/src/ggml-worker.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2025 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdlib.h>
+#include <pthread.h>
+#include <stdbool.h>
+#include <errno.h>
+#include <sched.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "ggml.h"
+#include "ggml-worker.h"
+
+static TaskQueue *g_queue[MAX_NUM_OF_QUEUES];
+static pthread_t g_threads[MAX_NUM_OF_THREADS];
+static int g_worker_initialized = false;
+static int g_num_of_threads;
+
+static void setSchedule(void)
+{
+  struct sched_param param;
+
+  memset(&param, 0, sizeof(param));
+  if (sched_getparam(0, &param) < 0) {
+      fprintf(stderr, "setSchedule: failed to sched_getparam errno:%d\n", errno);
+      return;
+  }
+
+  param.sched_priority = 1;
+  if (sched_setscheduler(0, SCHED_RR, &param) == -1) {
+      fprintf(stderr, "setSchedule: failed to sched_setscheduler(SCHED_RR) errno:%d\n", errno);
+      return;
+  }
```

Comment:
The `setSchedule` function attempts to set the scheduling policy to `SCHED_RR` (Round Robin) for both the main thread and worker threads.
Since `SCHED_RR` often requires elevated privileges, the call may fail if permissions are insufficient.
Check for failure and log an error.
Optionally, fallback to a more permissive scheduling policy (such as `SCHED_OTHER`) if `SCHED_RR` cannot be set.
```suggestion  
  param.sched_priority = 1;
  if (sched_setscheduler(0, SCHED_RR, &param) == -1) {
    fprintf(stderr, "setSchedule: failed to sched_setscheduler(SCHED_RR), errno: %d. Falling back to SCHED_OTHER.\n", errno);
    // SCHED_OTHER is the default scheduling policy on most systems
    if (sched_setscheduler(0, SCHED_OTHER, &param) == -1) {
        fprintf(stderr, "setSchedule: failed to set SCHED_OTHER as fallback, errno: %d\n", errno);
      }
    return;
  }
```

Reply:
> Optionally, fallback to a more permissive scheduling policy (such as SCHED_OTHER) if SCHED_RR cannot be set.

That's a good point. 
However, I think it would be better to keep the current policy if the RR change fails.